src/core/scope.ml

```diff
@@ -106,6 +106,17 @@ let rec get_implicitness : p_term -> bool list = fun t ->
   | P_Wrap(t)    -> get_implicitness t
   | _            -> []
 
+(** [_Meta_Type ~is_rhs env] create a new metavariable of type [TYPE] for a
+    missing domain in environment [env] in a RHS if [~is_rhs] is true. *)
+let _Meta_Type : is_rhs:bool -> env -> tbox = fun ~is_rhs env ->
+  let vs = Env.to_tbox env in
+  let a = Env.to_prod_box env _Type in
+  let m = _Meta_full (Meta.fresh_box a (Array.length vs)) vs in
+  (* Sanity check: only variables of [env] free in [m] if not in RHS. *)
+  if not is_rhs then
+    assert (Bindlib.is_closed (Bindlib.bind_mvar (Env.vars env) m));
```

Comment:
Do you think that it is really useful? I do not see the point of this check, and why not doing it if we are in a RHS? We should simply try to remove it and see whether the tests work.

Reply:
I tested: everything works.

Reply:
So _Meta_Type can be reduced to:
```
  let _Meta_Type : env -> tbox = fun env ->
    let vs = Env.to_tbox env in
    let a = Env.to_prod_box env _Type in
    _Meta_full (Meta.fresh_box a (Array.length vs)) vs
```

Reply:
This could be moved to libTerm:
```
(** Create a new metavariable of type [TYPE] in the environment [env]. *)
let _Meta_Type : env -> tbox = fun env ->
  let vs = to_tbox env in
  let arity = Array.length vs in
  let tm = to_prod_box env _Type in
  _Meta_full (Meta.fresh_box tm arity) vs

(** [fresh_meta_tbox env] creates a _Meta tbox from a fresh metavariable whose
   type is a fresh metavariable of type [to_prod env Type]. *)
let fresh_meta_tbox : env -> tbox = fun env ->
  let vs = to_tbox env in
  let arity = Array.length vs in
  let tm =
    let x = Meta.fresh_box (to_prod_box env _Type) arity in
    to_prod_box env (_Meta_full x vs)
  in
  _Meta_full (Meta.fresh_box tm arity) vs
```